File: backend/shared_utils/models/Post.py
from backend.shared_utils.models import Content

class Post(Content):
    def __init__(self, id, username, hashtags, text, likes, retweets, comments, source_url, timestamp, **kwargs):
        super().__init__(id=id, text=text, **kwargs)
        self.username = username
        self.hashtags = hashtags
        self.likes = likes
        self.retweets = retweets
        self.comments = comments  # List of Comment
        self.source_url = source_url
        self.timestamp = timestamp

File: frontend/lib/models/post.dart
import 'content.dart';
import 'comment.dart';

class Post extends Content {
  final String username;
  final String hashtags;
  final int likes;
  final int retweets;
  final List<Comment> comments;
  final String sourceURL;
  final DateTime timestamp;

  Post({
    required super.id,
    required this.username,
    required this.hashtags,
    required super.text,
    required this.likes,
    required this.retweets,
    required this.comments,
    required this.sourceURL,
    required this.timestamp,
    required super.isHumorous,
    required super.humorScore,
    required super.humorType,
    required super.topics,
    required super.tone,
    required super.emojiPresence,
    required super.textLength,
    required super.mediaType,
  });
}
